### DISCUSSION TITLE
Unable to change the logo using custom branding patch

### DISCUSSION THREAD

**Discussion by snoobysnoofy** [ReVanced/revanced-discussions#707] [Question]
I am unable to change logo of the app, here's how my whole directory looks like.
```
.
├── adb
├── app-release-unsigned.apk
├── options.toml
├── res
│   └── branding
│       ├── hdpi
│       │   ├── adaptiveproduct_youtube_background_color_108.png
│       │   ├── adaptiveproduct_youtube_foreground_color_108.png
│       │   ├── ic_launcher.png
│       │   └── ic_launcher_round.png
│       ├── mdpi
│       │   ├── adaptiveproduct_youtube_background_color_108.png
│       │   ├── adaptiveproduct_youtube_foreground_color_108.png
│       │   ├── ic_launcher.png
│       │   └── ic_launcher_round.png
│       ├── xhdpi
│       │   ├── adaptiveproduct_youtube_background_color_108.png
│       │   ├── adaptiveproduct_youtube_foreground_color_108.png
│       │   ├── ic_launcher.png
│       │   └── ic_launcher_round.png
│       ├── xxhdpi
│       │   ├── adaptiveproduct_youtube_background_color_108.png
│       │   ├── adaptiveproduct_youtube_foreground_color_108.png
│       │   ├── ic_launcher.png
│       │   └── ic_launcher_round.png
│       └── xxxhdpi
│           ├── adaptiveproduct_youtube_background_color_108.png
│           ├── adaptiveproduct_youtube_foreground_color_108.png
│           ├── ic_launcher.png
│           └── ic_launcher_round.png
├── revanced-cli-2.14.0-all.jar
├── revanced-patches-2.85.2.jar
├── vanced.apk
├── vanced.keystore
└── youtube.apk
```
I tried setting path in several different ways like,
`res/branding/`
`./res/branding/`
`res/`
`./res`

 but none of them worked, the patched apk is still with the original revanced icon.
```
# options.toml

['custom-branding']
appName = "Youtube Revanced"
appIconPath = "./res/"
```

How it should be done? I am unable to figure out this.
I am using the same command as mentioned in the docs
```
java -jar revanced-cli-2.14.0-all.jar \
 -a youtube.apk \
 -c \
 -d 330032c0a792c3fd \
 -o vanced.apk \
 -b revanced-patches-2.85.2.jar \
 -m app-release-unsigned.apk
```

**Comment by oSumAtrIX** [comment]
What exception did you get?

**Reply by snoobysnoofy** [reply]
sorry, I didn't noticed this earlier.
```
SEVERE: Error while setting option appIconPath for patch custom-branding: Invalid option value type: null, expected java.lang.Object 
app.revanced.patcher.patch.InvalidTypeException: Invalid option value type: null, expected java.lang.Object
	at app.revanced.utils.OptionsLoader.readAndSet(OptionsLoader.kt:82)
	at app.revanced.utils.OptionsLoader.init(OptionsLoader.kt:26)
	at app.revanced.cli.command.MainCommand.run(MainCommand.kt:137)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at app.revanced.cli.main.MainKt.main(Main.kt:7)
```

I am using Ubuntu 22.04
youtube app version is 17.36.37
and I am using latest release of all of the required tools revanced cli, patches and integrations
options.toml files says
`# This file was generated by ReVanced Patcher version 5.1.2.`

**Reply by ghost** [reply]
You need to use the full path.

**Reply by snoobysnoofy** [reply]
I also tried with full path,
`appIconPath = "/home/user/revanced/res/branding/"`

and other possible combinations like without `branding` without `/` etc.
`appIconPath = "/home/user/revanced/res"`
`appIconPath = "/home/user/revanced/res/"`
`appIconPath = "/home/user/revanced/res/branding"`
`appIconPath = "/home/user/revanced/"`

thinking that it searches for branding folder or res folder in the given directory.

its showing the same error as earlier

**Reply by E85Addict** [reply]
> I also tried with full path, `appIconPath = "/home/user/revanced/res/branding/"`
> 
> and other possible combinations like without `branding` without `/` etc. `appIconPath = "/home/user/revanced/res"` `appIconPath = "/home/user/revanced/res/"` `appIconPath = "/home/user/revanced/res/branding"` `appIconPath = "/home/user/revanced/"`
> 
> thinking that it searches for branding folder or res folder in the given directory.
> 
> its showing the same error as earlier

Did you get it too work?

**Reply by snoobysnoofy** [reply]
@E85Addict. yeah, but not by using the custom-branding patch.

I changed the logo from the source of revanced-patches which exists [here](https://github.com/revanced/revanced-patches/tree/main/src/main/resources/branding). then I compiled it locally on my machine, and used that to patch the apk.

**Comment by ghost** [comment]
Try it without using quotes.
